src/hotspot/share/cds/filemap.cpp

```diff
@@ -169,21 +170,44 @@ template <int N> static void get_header_version(char (&header_version) [N]) {
 FileMapInfo::FileMapInfo(bool is_static) {
   memset((void*)this, 0, sizeof(FileMapInfo));
   _is_static = is_static;
+  size_t c_header_size;
   size_t header_size;
+  size_t base_archive_name_size = 0;
+  size_t base_archive_path_offset = 0;
   if (is_static) {
     assert(_current_info == NULL, "must be singleton"); // not thread safe
     _current_info = this;
-    header_size = sizeof(FileMapHeader);
+    c_header_size = sizeof(FileMapHeader);
+    header_size = c_header_size;
   } else {
     assert(_dynamic_archive_info == NULL, "must be singleton"); // not thread safe
     _dynamic_archive_info = this;
-    header_size = sizeof(DynamicArchiveHeader);
+    // dynamic header including base archive name
+    c_header_size = sizeof(DynamicArchiveHeader);
+    header_size = c_header_size;
+    if (!FLAG_IS_DEFAULT(SharedArchiveFile)) {
+      base_archive_name_size = strlen(Arguments::GetSharedArchivePath()) + 1;
+      header_size += base_archive_name_size;
+    }
   }
   _header = (FileMapHeader*)os::malloc(header_size, mtInternal);
   memset((void*)_header, 0, header_size);
-  _header->set_header_size(header_size);
-  _header->set_version(INVALID_CDS_ARCHIVE_VERSION);
+  _header->set_base_archive_path_offset(0);
+  _header->set_base_archive_name_size(0);
+  _header->set_header_size((unsigned int)header_size);
   _header->set_has_platform_or_app_classes(true);
+  if (!is_static) {
+    _header->set_base_archive_is_default(FLAG_IS_DEFAULT(SharedArchiveFile));
```

Comment:
There's no need for a separate _base_archive_is_default field. We are using the base archive if base_archive_name_size == 0.